tensorflow_recommenders_addons/dynamic_embedding/core/kernels/lookup_impl/lookup_table_op_gpu.h

```diff
@@ -153,10 +159,7 @@ class TableWrapper final : public TableWrapperBase<K, V> {
                    d_dump_counter, stream);
       CUDA_CHECK(cudaStreamSynchronize(stream));
 
-      auto tmp_table_ = table_;
-      table_ = new Table(new_capacity);
-      delete tmp_table_;
-      tmp_table_ = NULL;
+      table_.reset(new Table(new_capacity));
```

Comment:
Is there any difference? The `reset` of `unique_ptr` will follow the order: construct a new table, then delete the old one. That means we had to prepare three times of HBM for this table. 
```cpp
auto tmp_table_ = table_.load();
auto tmp_table_new_ = new Table(new_capacity);
table_.store(tmp_table_new_ );
delete tmp_table_;

////
private:
  std::atomic<Table*> table_;
```